Controller/SubscriptionController.php

```diff
@@ -0,0 +1,291 @@
+                }
+            }
+            $newOrder->setUnivapayChargeAmount(json_encode($items));
+            $amount = json_decode($newOrder->getUnivapayChargeAmount());
+        }
+        $amount = get_object_vars($amount);
+        foreach($existOrder->getOrderItems() as $value) {
+            $newOrderItem = clone $value;
+            // 値引きは引き継がない
+            if($newOrderItem->isDiscount() || $newOrderItem->isPoint())
+                continue;
+            // OrderItemごとの金額を修正する
+            if($newOrderItem->isProduct()) {
+                // 二回目は通常金額を取得する
+                $class = $newOrderItem->getProductClass();
+                $newOrderItem->setPrice($amount[$class->getId()]->price);
+                $newOrderItem->setTax($amount[$class->getId()]->tax);
+            }
+            $newOrderItem->setOrder($newOrder);
+            $newOrder->addOrderItem($newOrderItem);
+        }
+        foreach($existOrder->getShippings() as $value) {
+            $newShipping = clone $value;
+            $newShipping->setShippingDeliveryDate(NULL);
+            $newShipping->setShippingDeliveryTime(NULL);
+            $newShipping->setTimeId(NULL);
+            $newShipping->setShippingDate(NULL);
+            $newShipping->setTrackingNumber(NULL);
+            $newShipping->setOrder($newOrder);
+            // 循環参照してしまい正常に発送データがセットできないため
+            foreach($newShipping->getOrderItems() as $v) {
+                $newShipping->removeOrderItem($v);
+            }
+            foreach($newOrder->getOrderItems() as $v) {
+                if($v->getShipping() && $v->getShipping()->getId() == $value->getId()) {
+                    $v->setShipping($newShipping);
+                    $newShipping->addOrderItem($v);
+                }
+            }
+            $newOrder->addShipping($newShipping);
+        }
+        $purchaseContext = new PurchaseContext($newOrder, $newOrder->getCustomer());
+        // ポイントを再計算4.0以前のバージョンの場合、先にポイントを再計算を行わないと正常に動かない
+        $this->addPointProcessor->validate($newOrder, $purchaseContext);
+        // 注文番号変更
+        $preOrderId = $this->orderHelper->createPreOrderId();
+        $newOrder->setPreOrderId($preOrderId);
+        // 購入処理を完了
+        $this->purchaseFlow->prepare($newOrder, $purchaseContext);
+        $this->purchaseFlow->commit($newOrder, $purchaseContext);
+        $this->entityManager->persist($newOrder);
+        // 注文番号が重複しないように再採番
+        $this->entityManager->flush();
+        $this->orderNoProcessor->process($newOrder, $purchaseContext);
+        $this->entityManager->flush();
+        // 定期課金に失敗した場合はキャンセル済み注文に変更
+        $OrderStatus = $this->orderStatusRepository->find($data->data->status === 'suspended' ? OrderStatus::CANCEL : OrderStatus::PAID);
+        $newOrder->setOrderStatus($OrderStatus);
+        if($config->getMail())
+            $this->mailService->sendOrderMail($newOrder);
+        $this->entityManager->flush();
+    }
```

Comment:
ここにレスポンスを返さなくてもいいですか？
```suggestion
        $this->entityManager->flush();
        return new Response()；
    }
```

Reply:
記載漏れでしたご指摘ありがとうございます 🙇